sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/SchemaTransform.java

```diff
@@ -33,5 +36,39 @@
  * compatibility guarantees and it should not be implemented outside of the Beam repository.
  */
 @Internal
-public abstract class SchemaTransform
-    extends PTransform<PCollectionRowTuple, PCollectionRowTuple> {}
+public abstract class SchemaTransform extends PTransform<PCollectionRowTuple, PCollectionRowTuple> {
+  private @Nullable Row configurationRow;
```

Comment:
It feels like putting these (optional) private variables as part of the base class violates separation of concerns.